SampleChangerMockup.py

```diff
@@ -23,23 +24,51 @@ def init(self):
         self.signal_wait_task = None
         SampleChanger.init(self)
 
-    def load_sample(self, holder_length, sample_location, wait):
-        return
+    def load_sample(self, holder_length, sample_location=None, wait=False):
+        self.load(sample_location, wait)
 
-    def load(self, sample, wait):
-        self._setState(SampleChangerState.Ready)  
-        return sample
+    def load(self, sample, wait=False):
+        try:
+            self._setState(SampleChangerState.Loading)
+            if isinstance(sample, tuple):
+                basket, sample = sample
+            else:
+                basket, sample = sample.split(":")
 
-    def unload(self, sample_slot, wait):
-        return
+            time.sleep(7)
+
+            self._setState(SampleChangerState.Ready)
+            self._triggerLoadedSampleChangedEvent(self.getLoadedSample())
+        except:
+            basket, sample = (None, None)
+            self._setState(SampleChangerState.Error)
+        finally:
+            self._selected_basket = int(basket)
+            self._selected_sample = int(sample)
 
+        return self.getLoadedSample()
+
+    def unload(self, sample_slot, wait):
+        self._selected_basket = None
+        self._selected_sample = None
+        self._triggerLoadedSampleChangedEvent(self.getLoadedSample())
+ 
     def getBasketList(self):
         basket_list = []
         for basket in self.components:
             if isinstance(basket, Basket):
                 basket_list.append(basket)
         return basket_list
 
+    def getLoadedSample(self):
```

Comment:
Hi @marcus-oscarsson 

getLoadedSample should return a Sample o Pin class and not a str. In the sample changer brick and dc_tree_widget it relies on this sample object.
I can revert to the previous version and add mockup load/unload logic.
Sorry for pointing this out after accepting the PR. I did not test it properly...